google/cloud/sql/connector/instance.py

```diff
@@ -356,10 +355,20 @@ async def _perform_refresh(self) -> InstanceMetadata:
                 )
             )
 
+            # gather Admin API tasks, return_exceptions=True to check auto iam authn
             metadata, ephemeral_cert = await asyncio.gather(
-                metadata_task, ephemeral_task
+                metadata_task, ephemeral_task, return_exceptions=True
             )
 
+            if isinstance(metadata, BaseException):
+                raise metadata
+
+            # check if automatic IAM database authn is supported for database engine
+            verify_iam_auth(metadata["database_version"], self._enable_iam_auth)
```

Comment:
Why can't this check be inlined? 

We can just check here and raise exception:

```python
if self._enable_iam_auth and not supports_iam_auth(metadata["database_version"]):
     raise Exception()
```